### PR TITLE
Enable nullability in Control.AmbientProperty, Control.Ime and Contro…

### PR DESCRIPTION
**Pull request by gpetrou**
…l.ThreadMethodEntry

## Proposed changes

- Enable nullability in Control.AmbientProperty, Control.Ime and Control.ThreadMethodEntry.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/4396)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.AmbientProperty.cs [hunk 1]
```diff
@@ -15,12 +13,12 @@ public partial class Control
...
         private class AmbientProperty
         {
-            private object _value;
+            private object? _value;
 
             /// <summary>
             ///  Creates a new, empty ambient property.
             /// </summary>
-            internal AmbientProperty(string name, Ole32.DispatchID dispID)
+            internal AmbientProperty(string? name, Ole32.DispatchID dispID)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.AmbientProperty.cs:21]: I don't think this is correct, there are only 3 callsites and none of them supply `null`.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ThreadMethodEntry.cs [hunk 2]
```diff
@@ -19,21 +17,21 @@ private class ThreadMethodEntry : IAsyncResult
...
-            internal ExecutionContext _executionContext;
+            internal ExecutionContext? _executionContext;
 
             // Optionally store the synchronization context associated with the callee thread.
             // This overrides the sync context in the execution context of the caller thread.
-            internal SynchronizationContext _syncContext;
+            internal SynchronizationContext? _syncContext;
 
-            internal ThreadMethodEntry(Control caller, Control marshaler, Delegate method, object[] args, bool synchronous, ExecutionContext executionContext)
+            internal ThreadMethodEntry(Control caller, Control marshaler, Delegate method, object[] args, bool synchronous, ExecutionContext? executionContext)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.ThreadMethodEntry.cs:34]:
```suggestion
            internal ThreadMethodEntry(Control caller, Control marshaler, Delegate? method, object[]? args, bool synchronous, ExecutionContext? executionContext)
```

image
image

File: src/System.Windows.Forms/src/PublicAPI.Shipped.txt [hunk 3]
```diff
@@ -839,7 +839,7 @@ System.Windows.Forms.Control.HelpRequested -> System.Windows.Forms.HelpEventHand
 System.Windows.Forms.Control.Hide() -> void
 System.Windows.Forms.Control.ImeMode.get -> System.Windows.Forms.ImeMode
 System.Windows.Forms.Control.ImeMode.set -> void
-System.Windows.Forms.Control.ImeModeChanged -> System.EventHandler
+System.Windows.Forms.Control.ImeModeChanged -> System.EventHandler!
```
Comment (RussKie) [inline src/System.Windows.Forms/src/PublicAPI.Shipped.txt:842]: This is a non-obvious change. How did we arrive to it?

**Review by RussKie — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/src/PublicAPI.Shipped.txt (hunk 3 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/PublicAPI.Shipped.txt:842]: I could not find cases where it could be null. Should I change it?

**Review by RussKie — DISMISSED** [review]

**Comment by RussKie** [comment]
Please rebase and resolve the conflict.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**